### ISSUE TITLE
Test failures on Alpine Linux

### ISSUE THREAD

**Issue by orhun** [prefix-dev/pixi#821]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

I'm packaging `pixi` for Alpine Linux and I hit a test failure. You can use the [`APKBUILD` from this PR](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/61116) and use `abuild -r` command to build it.

### Issue description

```
failures:
---- task::task_environment::tests::test_find_ambiguous_task stdout ----
thread 'task::task_environment::tests::test_find_ambiguous_task' panicked at src/task/task_environment.rs:337:9:
assertion failed: matches!(result, Err(FindTaskError::AmbiguousTask(_)))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---- task::task_environment::tests::test_find_task_dual_defined stdout ----
thread 'task::task_environment::tests::test_find_task_dual_defined' panicked at src/task/task_environment.rs:246:9:
assertion failed: matches!(result, Err(FindTaskError::AmbiguousTask(_)))
---- task::task_environment::tests::test_find_task_explicit_defined stdout ----
thread 'task::task_environment::tests::test_find_task_explicit_defined' panicked at src/task/task_environment.rs:272:9:
assertion failed: matches!(result, Err(FindTaskError::AmbiguousTask(_)))
failures:
    task::task_environment::tests::test_find_ambiguous_task
    task::task_environment::tests::test_find_task_dual_defined
    task::task_environment::tests::test_find_task_explicit_defined
```

I applied a patch to make it more clear - the tests expect `Err` but the result is actually `Ok`:

```
---- task::task_environment::tests::test_find_ambiguous_task stdout ----
thread 'task::task_environment::tests::test_find_ambiguous_task' panicked at src/task/task_environment.rs:337:31:
called `Result::unwrap_err()` on an `Ok` value: (Environment { project: "foo", environment: Default }, Plain("echo foo"))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- task::task_environment::tests::test_find_task_explicit_defined stdout ----
thread 'task::task_environment::tests::test_find_task_explicit_defined' panicked at src/task/task_environment.rs:272:31:
called `Result::unwrap_err()` on an `Ok` value: (Environment { project: "foo", environment: Default }, Plain("pytest -s"))

---- task::task_environment::tests::test_find_task_dual_defined stdout ----
thread 'task::task_environment::tests::test_find_task_dual_defined' panicked at src/task/task_environment.rs:246:31:
called `Result::unwrap_err()` on an `Ok` value: (Environment { project: "foo", environment: Default }, Plain("cargo test"))
failures:
    task::task_environment::tests::test_find_ambiguous_task
    task::task_environment::tests::test_find_task_dual_defined
    task::task_environment::tests::test_find_task_explicit_defined
```

CI status: https://gitlab.alpinelinux.org/orhun/aports/-/jobs/1283156

### Expected behavior

Passing tests

**Comment by benglewis** [comment]
You can install `glibc` from https://github.com/sgerrand/alpine-pkg-glibc , however this does not seem to help resolve the issues. Is there a particular reason that `glibc` is needed and that `musl libc` cannot be used?

**Comment by ruben-arts** [comment]
@benglewis It's not a dependency of pixi but a dependency of the `conda-forge` packages. They're build against `glibc` and not `musl libc`. So it would require a rebuild with that as a base dependency. I believe it would be possible to make a line of packages on conda forge which are build for musl.

**Comment by benglewis** [comment]
@ruben-arts Got it, but while `micromamba` works fine with `conda-forge` on `alpine`, somehow `prefix` does not. So it seems like the `sgerrand/alpine-pkg-glibc` does work for `conda-forge` packages, but for some reason, it does not work with `pixi`

**Comment by ruben-arts** [comment]
Oh that is indeed not what I would expect. In theory the following two scripts should result in 99.9% the same environment:
```
pixi init test && cd test
pixi add python
pixi run python
```

```
micromamba create -n test -c conda-forge python
micromamba acitvate test
python
```

You could compare the two results. 

If this doesn't work. You could share the `pixi info` output. 
It should pickup a `__glibc` version. 
If not then our virtual package detection system might need to be extended.

**Comment by benglewis** [comment]
Sure, here's the output of the `pixi info`:
```
      Pixi version: 0.21.1
          Platform: linux-64
  Virtual packages: __unix=0=0
                  : __linux=6.6.26=0
                  : __archspec=1=x86_64
         Cache dir: /root/.cache/rattler/cache
      Auth storage: /root/.rattler/credentials.json
```
So evidently, `__glibc` isn't found properly.

However... I just found a solution, yay! 🎆 
It seems like the issue I was facing was due to the base Docker image (namely, `frolvlad/alpine-glibc:alpine-3.19`). Switching to `woahbase/alpine-glibc` resolved the issue. It isn't clear to me what exactly are the different things that they do. I don't know if it is related to the `PATH` environment variable or something else to do with how they compile things. But sweet!

I share this here, in the hope that it will help you test `pixi` on this base image yourselves :)
